docs/project-config.md

```diff
@@ -0,0 +1,279 @@
+
+
+### Configuration
+
+For each CLI plugin, you can add configurations as an array of options that you pass to the plugin when you instantiate it:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      // Configuration
+    })
+  ],
+});
+```
+
+Each plugin has its own set of options that you can find on the respective documentation pages.
+
+
+### File Builder
+
+For each CLI plugin that hooks into the `build` command, you can use the Jovo File Builder to write specific files into a path of the `build` directory.
+
+You can add the paths and file content to the `files` property of the plugin config:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      files: {
+        path: {
+          to: {
+            'file.json': 'Hello World!'
+          }
+        }
+      }
+    })
+  ],
+});
+```
+
+The above example creates a file in `build/alexa/path/to/file.json` with the content `Hello World!`.
+
+You can also use this shorthand for nested folders:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      files: {
+        'path.to.file.json': 'Hello World!'
+      }
+    })
+  ],
+});
+```
+
+
+## Staging
+
+Project staging allows you to `build` for and `deploy` to specific staging environments. A common use case is to have a `dev` stage for local development and a `prod` stage hosted in the cloud.
+
+Here is an example, how this could look like:
+
+```js
+const project = new ProjectConfig({
+  // ...
+
+  defaultStage: 'dev',
+  stages: {
+    dev: {
+      endpoint: '${JOVO_WEBHOOK_URL}',
+      // ...
+    },
+    prod: {
+      endpoint: process.env.ENDPOINT_PROD,
+      // ...
+    }
+  }
+});
+```
+
+In the example above, each stage has its own [`endpoint`](#endpoint). The `dev` stage uses the Jovo Webhook URL, and the `prod` stage a URL to a hosted service that is provided via environment variables in a `.env` file.
+
+You can add any stage name to the `stages` object:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  stages: {
+    someStage: {
+      // ...
+    }
+  }
+});
+```
+
+Inside this stage, you can add any configuration that you can also add outside a stage, even `plugins`:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  stages: {
+    someStage: {
+      // ...
+      plugins: [
+        new AlexaCli({ skillId: 'someSkillId' });
+      ]
+    }
+  }
+});
+```
+
+By default, only the elements outside the `stages` are used. If a specific stage is active, all content from this stage is merged into the generic config.
+
+The active stage is determined in the following order (most active first):
+
+* The stage is added as flag, e.g. `jovov4 build --stage someStage`
+* The default stage is set with `defaultStage: 'someStage'`
+* The stage is set in the environment variables with `NODE_ENV=someStage`
```

Comment:
Currently, the implementation takes the `defaultStage` as the last option. The following order is correct:

1. Flag, e.g. `jovov4 build --stage someStage`
2. `process.env.JOVO_STAGE`
3. `process.env.NODE_ENV`
4. `defaultStage: someStage`

Is this something we should discuss further?

Reply:
Ah yes, this makes sense! Thank you, I'll update